src/ddsrt/src/string.c

```diff
@@ -219,3 +219,41 @@ ddsrt_strndup(const char *str, size_t len)
 
   return s;
 }
+
+void
+ddsrt_str_trim(char *str)
+{
+  int index, i;
```

Comment:
The other string functions consistently use `size_t` to index the string. Given how they are input, I doubt the distinguished names can ever be > 2GB, but using `int` here means there is a limit that one would not expect based on the other string functions.